### DISCUSSION TITLE
Errors during render phase on iOS

### DISCUSSION THREAD

**Discussion by Gruski** [mozilla/pdf.js#17669] [Q&A]
I am rendering a document with larger pages like blueprints and API starts rendering pages (one canvas per page) then craps out during the canvas render phase after a few pages on all iPhones. Even with normal 8.5/11 size pages it prints more pages but after 7 or 10 same problem. The lower the scale resolution, the more pages get rendered before failure. Seems like a processing scale problem (memory problem?) Firefox/Chrome on desktop and Android Phone have no problem with any amount of pages of any size and cranks them out with lightning speed.

Are there any special considerations that need to be taken into account with Safari or on iPhones to get this API working?

**Comment by oddcb** [comment]
Hi. If you debug the app and look at the browser console logs, do you get something that says that the canvas memory limit has been exceeded? The symptom in the pdf.js would probably be that it loads the document, but displays a blank page.

You can look into https://bugs.webkit.org/show_bug.cgi?id=195325 but it seems that the fix only landed in Safari 17. In older Safari (at least 15 and 16 from my testing) it seems that the memory limit is (allocable memory for application / 4).
